packages/dx-chart-core/src/utils/scale.ts

```diff
@@ -11,9 +11,12 @@ import {
 /** @internal */
 export const scaleLinear: FactoryFn = d3ScaleLinear as any;
 /** @internal */
-export const scaleBand: FactoryFn = () => (
-  d3ScaleBand().paddingInner(0.3).paddingOuter(0.15) as any
-);
+export const scaleBand: FactoryFn = () => {
+  const scale = d3ScaleBand().paddingInner(0.3).paddingOuter(0.15);
+  const wrapper = (value: any) => scale(value)! + scale.bandwidth() / 2;
+  Object.assign(wrapper, scale);
+  return wrapper as ScaleObject;
```

Comment:
Why we don't use new syntax?

```js
return { ...wrapper, ...scale }
```

Reply:
That's not the same - `assign` copies all keys from `scale` to `wrapper`, spread copies all keys from `wrapper` and `scale` to a new object.

Reply:
But a function result is same

Reply:
How can it be the same?

Reply:
My mistake, I didn't see that wrapper is a function.